packages/core/src/new-api/internal/execution/transactions/chain-dispatcher.ts

```diff
@@ -2,11 +2,19 @@ import { ethers } from "ethers";
 
 import { GasAdapter, TransactionsAdapter } from "../../../types/adapters";
 
-interface TransactionReceipt {
+interface TransactionSuccess {
+  type: "transaction-success";
   contractAddress?: string;
   txId?: string;
 }
 
+interface TransactionFailure {
+  type: "transaction-failure";
+  error: Error;
+}
+
+type TransactionReceipt = TransactionSuccess | TransactionFailure;
```

Comment:
I think this type is really confusing. `TransactionReceipt` as receipt is already used in the protocol.

Reply:
This is a terrible name. But that code is getting folded (deleted) into the execution-engine.